src/store/entities/Member.java

```diff
@@ -37,13 +34,13 @@
 	 *                    today's date to the member; the date is assigned by user
 	 * @param feePaid     - membership fee the new member paid
 	 */
-	public Member(String name, String address, String phoneNumber, Calendar dateJoined, double feePaid) {
+	public Member(String name, String address, String phoneNumber, Calendar dateJoined, double feePaid, int idCounter) {
```

Comment:
Why are we having the id counter iterate in the grocery store facade and not the class?

Reply:
In order to Serialize it I need a direct access to the counter from GroceryStore. But it was buried down in Member and Order. (Not even 1 level up, in MembersList and OrdersList. I couldn't reach it to save or load it. If it wasn't for the serialization, it would've been fine. They are still private fields, we just have to pass them down. I couldn't think of a more elegant solution.